web_widget_date_interval/static/src/js/web_widget_date_interval.js

```diff
@@ -12,117 +14,114 @@ openerp.web_widget_date_interval = function(instance)
-            {
+            _get_date: function () {
                 return instance.web.str_to_date(
                     instance.web.parse_value(
                         this.get_value(), this, Date.now()
                     ).substring(0, 10)
                 );
             },
-            weeknumber_iso_weeknumber: function(date)
-            {
-                // getISOWeek looks at the UTC time, but users expect walltime
+            weeknumber_iso_weeknumber: function (date) {
+                // GetISOWeek looks at the UTC time, but users expect walltime
                 var local = date.clone().clearTime().add(12).hours();
-                return parseInt(local.getISOWeek());
+                return parseInt(local.getISOWeek(), 10);
             },
-            weeknumber_iso_get_years: function()
-            {
-                if(this.options.hide_years)
-                {
+            weeknumber_iso_get_years: function () {
+                if (this.options.hide_years) {
                     return [];
                 }
                 var result = [],
                     years_before = this.options.years_before || 5,
                     years_after = this.options.years_after || 5,
                     current_date = this._get_date();
-                for(
+                for (
                     var year=current_date.getFullYear() - years_before;
                     year <= current_date.getFullYear() + years_after;
                     year++
-                )
-                {
+                ) {
                     result.push(year);
                 }
                 return result;
             },
-            weeknumber_iso_get_weeks: function()
-            {
+            weeknumber_iso_get_weeks: function () {
                 var result = [],
                     last_day = this._get_date(),
                     max_week = 52;
-                if(!last_day.is().december())
-                {
+                if (!last_day.is().december()) {
                     last_day.december();
                 }
                 last_day.moveToLastDayOfMonth();
-                if(this.weeknumber_iso_weeknumber(last_day) > 1)
-                {
+                if (this.weeknumber_iso_weeknumber(last_day) > 1) {
                     max_week = this.weeknumber_iso_weeknumber(last_day);
                 }
-                for(var week=1; week <= max_week; week++)
-                {
+                for (var week=1; week <= max_week; week++) {
                     result.push(week);
                 }
                 return result;
             },
-            weeknumber_iso_onchange_year: function(e)
-            {
-                var year = parseInt(jQuery(e.currentTarget).data('year')),
-                    value = this._get_date();
-                value.setFullYear(year);
-                return this.weeknumber_iso_update_interval(value);
+            get_selected_year_and_week: function (e) {
```

Comment:
this is conceptually wrong. Not the DOM saves values, the dataset does. If for whatever reason the value in the dataset is wrong, you need to investigate why, not just circumvent it